pkg/operator/utils/utils.go

```diff
@@ -147,30 +149,102 @@ func GenerateNameWithFieldLimits(infraName string, infraNameMaxLen int, crName s
 	return fmt.Sprintf("%s%s", infraPrefix, crName), nil
 }
 
-// IsOperatorDisabled checks the cloud-credential-operator-config ConfigMap for a
+// IsOperatorDisabledViaConfigmap checks the cloud-credential-operator-config ConfigMap for a
 // "disabled" property set to true. If the configmap or property do not exist, we assume
 // false and continue normal operation. This should be used in all controllers to shutdown
 // functionality in environments where admins want to manage their credentials themselves.
-func IsOperatorDisabled(kubeClient client.Client, logger log.FieldLogger) (bool, error) {
+// DEPRECATED (and unexported), use GetOperatorConfiguration to determine disabled state.
+func isOperatorDisabledViaConfigmap(kubeClient client.Client, logger log.FieldLogger) (bool, error) {
+	cm, err := GetLegacyConfigMap(kubeClient)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Debugf("%s ConfigMap does not exist, assuming default behavior", constants.CloudCredOperatorConfigMap)
+			return OperatorDisabledDefault, nil
+		}
+		return OperatorDisabledDefault, err
+	}
+
+	return CCODisabledCheck(cm, logger)
+}
+
+func GetLegacyConfigMap(kubeClient client.Client) (*corev1.ConfigMap, error) {
 	cm := &corev1.ConfigMap{}
 	err := kubeClient.Get(context.TODO(),
 		types.NamespacedName{
 			Namespace: minterv1.CloudCredOperatorNamespace,
 			Name:      constants.CloudCredOperatorConfigMap,
 		}, cm)
+	return cm, err
+}
+
+// GetOperatorConfiguration will return the value in the operator config, and whether the operator has been disabled
+// via either the "manual" mode setting, or the legacy configmap, and whether there is a conflict between the legacy
+// and CCO config (in the even of a conflict, the disabledSetting field will reflect the value in the operator config
+// in the event of a conflict).
+func GetOperatorConfiguration(kubeClient client.Client, logger log.FieldLogger) (
+	operatorMode operatorv1.CloudCredentialsMode,
+	disabledSetting bool,
```

Comment:
I still don't think there is much benefit is returning a separate bool for disabled. The caller can easily check `operatorMode == ManualMode`.

Reply:
If operatorMode == "" the legacy configmap might still have an opinion about disabled/manual mode.

I think I ended up organizing everything behind this GetOperatorConfiguration() b/c I was originally returning an error when there was a conflict between the configmap and the operator config, and the status handlers needed a way to set conditions based on the various states that we can find ourselves in.

Reply:
My contention is that the `operatorMode` returned should be "manual" if the config CR has not mode set and the disabled ConfigMap is set to disabled.

Reply:
It does not matter to callers of `GetOperatorConfiguration` whether the mode was set via the config CR or via the disabled CofigMap.

Reply:
Fair enough. I dropped the disabled field, and will just insert the "manual" mode if the conditions are right.